go/vt/vtgate/semantics/dependencies.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2021 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package semantics
+
+import (
+	querypb "vitess.io/vitess/go/vt/proto/query"
+	vtrpcpb "vitess.io/vitess/go/vt/proto/vtrpc"
+	"vitess.io/vitess/go/vt/vterrors"
+)
+
+type (
+	dependencies interface {
+		Empty() bool
+		Get() (dependency, error)
+		Merge(dependencies) (dependencies, error)
+	}
+	dependency struct {
+		direct    TableSet
+		recursive TableSet
```

Comment:
Probably in a separate PR: we need to make sure we change the names of our dependencies from `base` to `recursive`, and `real|deps` to `direct` throughout the whole codebase

Reply:
Resolved via eb1e967